emcc.py

```diff
@@ -1458,6 +1458,8 @@ def phase_setup(options, state, newargs, user_settings):
   if settings.USE_PTHREADS or settings.WASM_WORKERS:
     settings.SHARED_MEMORY = 1
 
+  newargs += ['-mmutable-globals']
```

Comment:
This represents to a much larger change since it effect all object files that are built by emscripten.   I'm not sure we want to this right now just for the benefit of `STACK_OVERFLOW_CHECK=2`.    I think I prefer the previous approach since it doesn't effect object files and it doesn't effect builds that are not using `STACK_OVERFLOW_CHECK=2`